rfd/0018-centos-6-eol.md

```diff
@@ -0,0 +1,63 @@
+---
+authors: Gus Luxton (gus@goteleport.com)
+state: draft
+---
+
+# RFD 18 - EOL for Teleport CentOS 6 binaries
+
+## What
+
+The Teleport build pipeline currently builds special CentOS 6 binaries (which are linked
+against glibc 2.12) for customers who have not yet upgraded from CentOS 6.
+
+CentOS 6 was originally released in July 2011. Full updates for all its packages stopped
+in May 2017, and [CentOS 6 has been officially declared EOL](https://wiki.centos.org/FAQ/General#What_is_the_support_.27.27end_of_life.27.27_for_each_CentOS_release.3F)
+as of November 2020.
+
+The [CentOS 6 official package mirrors have been taken down](http://mirror.centos.org/centos-6/6/readme)
+and the message being shared here by the maintainers could not be clearer - `"The whole CentOS 6 is *dead* and *shouldn't* be used anywhere at *all*"`
+
+As such, we will no longer produce CentOS 6 binaries for the next major version of Teleport (6.0).
+
+## Why
+
+Using EOL distros is unsafe and even dangerous for many reasons. CVEs and security bugs are regularly found
+and never fixed, leaving servers insecure and easily pwned by trivial exploits.
+
+Continuing to support CentOS 6 is undesirable:
+- Teleport is fundamentally positioned as a security product - its objective is to provide secure access
+  to infrastructure. Continuing to build binaries for an end-of-life distribution sends a bad message;
+  that Teleport is willing to support the use of end-of-life distributions which do not receive any
+  security updates.
+- Having separate special-cased builds increases the size and complexity of Teleport's build
+  matrix, often requiring workarounds to support features which newer distributions provide out of the box.
+- Some Teleport features (for example: enhanced session recording) do not work on CentOS 6 because
+  the kernels used are too old. Inconsistencies in the Teleport experience across distributions are
+  bad for product messaging and potential customers.
```

Comment:
Same could be said for macOS and Linux/ARM.

Reply:
Yes, but people aren't generally running MacOS servers and enhanced session recording does work on ARM.